By me: Added partamater to center window when width is set
---
diff --git a/dmenu.c b/dmenu.c
index f54a6f0..8925a39 100644
--- a/dmenu.c
+++ b/dmenu.c
@@ -56,6 +56,7 @@ static int mon = -1, screen;
 static int lines_below_prompt = 0;
 static int password_height_offset = 0;
 static int no_options = 0;
+static int center = 0;
 
 static Atom clip, utf8;
 static Display *dpy;
@@ -766,6 +767,10 @@ setup(void)
 				if (INTERSECT(x, y, 1, 1, info[i]))
 					break;
 
+		if (center && dmw > 0) {
+		    dmx = (info[i].width - dmw) / 2;
+		}
+
 		x = info[i].x_org + dmx;
 		y = info[i].y_org + (topbar ? dmy : info[i].height - mh - dmy);
 		mw = (dmw>0 ? dmw : info[i].width);
@@ -819,7 +824,7 @@ static void
 usage(void)
 {
 	fputs("usage: dmenu [-bfivP] [-l lines] [-h height] [-p prompt] [-fn font] [-m monitor]\n"
-	      "             [-x xoffset] [-y yoffset] [-z width]\n"
+	      "             [-x xoffset] [-y yoffset] [-z width] [-c]\n"
 	      "             [-lbp] [-no]\n"
 	      "             [-nb color] [-nf color] [-sb color] [-sf color] [-w windowid]\n"
 	      "             [-d separator] [-D separator]\n", stderr);
@@ -852,6 +857,8 @@ main(int argc, char *argv[])
 			passwd = 1;
 		else if (!strcmp(argv[i], "-no"))  /* disable showing options (input only) */
 			no_options = 1;
+		else if (!strcmp(argv[i], "-c"))  /* centers window (overrides -x) */
+			center = 1;
 		else if (i + 1 == argc)
 			usage();
 		/* these options take one argument */
